sql/analyzer/resolve_ctes.go

```diff
@@ -142,11 +142,7 @@ func stripWith(ctx *sql.Context, a *Analyzer, scope *Scope, n sql.Node, ctes map
 		subquery := cte.Subquery
 
 		if len(cte.Columns) > 0 {
-			schemaLen := schemaLength(subquery)
-			if schemaLen != len(cte.Columns) {
-				return nil, sql.ErrColumnCountMismatch.New()
-			}
-
+			// wait until we've resolved stars to error for column mismatch
```

Comment:
This comment really only refers to code that was deleted

A better one might be `We don't validate the number of columns in the CTE schema until later, see (code ref)`